### PR TITLE
Fix 627

### PR DESCRIPTION
**Pull request by zhxfl**
split long sentence, drop sentence close by -1

### CODE REVIEW

File: fluid/DeepASR/data_utils/data_reader.py [hunk 1]
```diff
@@ -99,12 +111,42 @@ def generate_sample_info_list(self):
...
                 label_frame_num = int(label_desc_split[4])
-
-                sample_info_list.append(
-                    SampleInfo(feature_bin_path, feature_start, feature_size,
-                               feature_frame_num, feature_dim, label_bin_path,
-                               label_start, label_size, label_frame_num))
-
+                assert feature_frame_num == label_frame_num
+
+                if self._split_sentence_threshold == -1 or self._split_perturb == -1 or self._split_sub_sentence_len == -1 or self._split_sentence_threshold >= feature_frame_num:
```
Comment (kuke) [inline fluid/DeepASR/data_utils/data_reader.py]: Please keep one line under the 80-character limit, which is also suitable for comments

Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: fix

**Comment by kuke** [comment]
LGTM

**Comment by kuke** [comment]
LGTM

File: fluid/DeepASR/data_utils/data_reader.py [hunk 2]
```diff
@@ -61,10 +61,21 @@ class SampleInfoBucket(object):
         label_bin_paths (list|tuple): Files containing the binary label data.
         label_desc_paths (list|tuple): Files containing the description of
                                        samples' label data.
+        split_perturb(int): split long sentence' perturb sub-sentence length value. 
+        split_sentence_threshold(int): sentence length large than
```
Comment (pkuyym) [inline fluid/DeepASR/data_utils/data_reader.py]: sentence --> Sentence
large --> larger
operator --> operation

sentence length large than split_sentence_threshold trigger split operator. -->
Sentence whose length larger than the value will trigger split operation.

File: fluid/DeepASR/data_utils/data_reader.py [hunk 3]
```diff
@@ -61,10 +61,21 @@ class SampleInfoBucket(object):
         label_bin_paths (list|tuple): Files containing the binary label data.
         label_desc_paths (list|tuple): Files containing the description of
                                        samples' label data.
+        split_perturb(int): split long sentence' perturb sub-sentence length value.
```
Comment (pkuyym) [inline fluid/DeepASR/data_utils/data_reader.py]: split_perturb(int) --> split_perturb (int)

Comment (pkuyym) [reply fluid/DeepASR/data_utils/data_reader.py]: `split long sentence' perturb sub-sentence length value.` Please refine this comment.

File: fluid/DeepASR/data_utils/data_reader.py [hunk 4]
```diff
@@ -101,12 +115,45 @@ def generate_sample_info_list(self):
...
+                                   label_bin_path, label_start, label_size,
+                                   label_frame_num))
+                #split sentence 
+                else:
+                    cur_frame_pos = 0
+                    cur_frame_len = 0
+                    remain_frame_num = feature_frame_num
+                    while True:
+                        if remain_frame_num > self._split_sentence_threshold:
+                            cur_frame_len = self._split_sub_sentence_len + random.randint(
```
Comment (pkuyym) [inline fluid/DeepASR/data_utils/data_reader.py]: Seems exceed 80 columns.

File: fluid/DeepASR/data_utils/data_reader.py [hunk 5]
```diff
@@ -244,11 +291,20 @@ def read_bytes(fpath, start, size):
                                            sample_info.feature_start,
                                            sample_info.feature_size)
 
+                assert sample_info.feature_frame_num * sample_info.feature_dim * 4 == len(
```
Comment (pkuyym) [inline fluid/DeepASR/data_utils/data_reader.py]: Seems exceed 80 columns.

File: fluid/DeepASR/data_utils/data_reader.py [hunk 6]
```diff
@@ -273,7 +329,8 @@ def read_bytes(fpath, start, size):
                     time.sleep(0.001)
 
                 # drop long sentence
-                if self._drop_frame_len >= sample_data[0].shape[0]:
+                if self._drop_frame_len == -1 or self._drop_frame_len >= sample_data[
```
Comment (pkuyym) [inline fluid/DeepASR/data_utils/data_reader.py]: Same as above.

**Review by pkuyym — CHANGES_REQUESTED** [review]

File: fluid/DeepASR/data_utils/data_reader.py (hunk 4 above)
Comment (pkuyym) [reply fluid/DeepASR/data_utils/data_reader.py]: Please use `self._rng` instead using `random.randint` directly.

File: fluid/DeepASR/data_utils/data_reader.py (hunk 3 above)
Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: Random perturb sub-sentence length when split long sentence

File: fluid/DeepASR/data_utils/data_reader.py (hunk 2 above)
Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: OK

File: fluid/DeepASR/data_utils/data_reader.py (hunk 4 above)
Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: self._rng.randint(0, self._split_perturb)

File: fluid/DeepASR/data_utils/data_reader.py (hunk 5 above)
Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: agree

File: fluid/DeepASR/data_utils/data_reader.py (hunk 6 above)
Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: fix

File: fluid/DeepASR/data_utils/data_reader.py (hunk 3 above)
Comment (pkuyym) [reply fluid/DeepASR/data_utils/data_reader.py]: How about `Maximum perturbation value for length of sub-sentence when splitting long sentence.`?

Comment (zhxfl) [reply fluid/DeepASR/data_utils/data_reader.py]: nice

**Review by pkuyym — APPROVED** [review]
LGTM

**Pull request merged**